admin/src/main/java/edu/unc/lib/dl/admin/collect/IngestSourceManager.java

```diff
@@ -0,0 +1,303 @@
+				PID containerPID = new PID(container);
+				if (containerPID.equals(destination) || ancestors.contains(containerPID)) {
+					applicableSources.add(source);
+					continue;
+				}
+			}
+		}
+
+		return applicableSources;
+	}
+
+	/**
+	 * Retrieves a list of candidate file information for ingestable packages from sources which are
+	 * applicable to the destination provided.
+	 * 
+	 * @param destination
+	 * @return
+	 */
+	public List<Map<String, Object>> listCandidates(PID destination) {
+
+		List<IngestSourceConfiguration> applicableSources = listSources(destination);
+
+		final List<Map<String, Object>> candidates = new ArrayList<>();
+		for (final IngestSourceConfiguration source : applicableSources) {
+			final String base = source.getBase();
+			
+			// Gathering candidates per pattern within a particular base directory
+			for (String pattern : source.getPatterns()) {
+				final PathMatcher matcher = FileSystems.getDefault().getPathMatcher("glob:" + base + pattern);
+				try {
+					Files.walkFileTree(Paths.get(base), new SimpleFileVisitor<Path>() {
+						@Override
+						public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+							if (matcher.matches(dir)) {
+								log.debug("Matched dir {} for source {}", dir, source.getId());
+								addCandidate(candidates, dir, source, base);
+								return FileVisitResult.SKIP_SUBTREE;
+							}
+							return FileVisitResult.CONTINUE;
+						}
+						
+						@Override
+						public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+							if (matcher.matches(file)) {
+								log.debug("Matched file {} for source {}", file, source.getId());
+								addCandidate(candidates, file, source, base);
+							}
+							return FileVisitResult.CONTINUE;
+						}
+					});
+				} catch (IOException e) {
+					log.error("Failed to gather candidate files for source {}", source.getId(), e);
+				}
+			}
+		}
+		
+		return candidates;
+	}
+
+	/**
+	 * Adds information about applicable packages to the list of candidates
+	 * 
+	 * @param candidates
+	 * @param filePath
+	 * @param source
+	 * @param base
+	 * @throws IOException
+	 */
+	private void addCandidate(List<Map<String, Object>> candidates, Path filePath,
+			IngestSourceConfiguration source, String base) throws IOException {
+		
+		File file = filePath.toFile();
+		// Only directory bags are candidates currently
+		String version = BagHelper.getVersion(file);
+		if (version == null || !file.isDirectory()) {
+			return;
+		}
+		
+		Map<String, Object> candidate = new HashMap<>();
+		
+		candidate.put("sourceId", source.getId());
+		candidate.put("base", base);
+		candidate.put("patternMatched", Paths.get(base).relativize(filePath).toString());
+		
+		candidate.put("version", version);
+
+		if (version != null) {
+			// Add payload stats for bags
+			addBagInfo(candidate, filePath);
+		}/* else if (file.isDirectory()) {
```

Comment:
Yes, its dead code currently, but we have a ticket for adding deposit from plain directories so I figured we might as well keep it until we get to that, although perhaps it'd be better to move it to a branch related to that purpose